api/v1beta1/module_webhook_test.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"testing"
+)
+
+func TestV1beta1(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "V1beta1 Suite")
+}
+
+var _ = Describe("validateKernelMappingRegex", func() {
+	It("should pass when there are not kernel mappings", func() {
+		e := (&Module{}).validateKernelMappingRegex()
+		Expect(e).ToNot(HaveOccurred())
+	})
+
+	It("should pass when regexp and containerImage are valid", func() {
+		mod1 := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						KernelMappings: []KernelMapping{
+							{Regexp: "", ContainerImage: "image-url"},
+						},
+					},
+				},
+			},
+		}
+
+		mod2 := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						ContainerImage: "image-url",
+						KernelMappings: []KernelMapping{
+							{Regexp: "valid-regex"},
+						},
+					},
+				},
+			},
+		}
+
+		Expect(mod1.validateKernelMappingRegex()).ToNot(HaveOccurred())
+		Expect(mod2.validateKernelMappingRegex()).ToNot(HaveOccurred())
+	})
+
+	It("should fail when an invalid regex is found", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						KernelMappings: []KernelMapping{
+							{Regexp: "*-invalid-regex"},
+						},
+					},
+				},
+			},
+		}
+
+		e := mod.ValidateCreate()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(ContainSubstring("invalid regexp"))
+	})
+
+	It("should fail when a kernel-mapping has invalid containerName", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						Modprobe: ModprobeSpec{
+							ModuleName: "mod-name",
+						},
```

Comment:
Is this relevant to the test? I think the code will fail before reaching that check anyway.